src/vcpkg/commands.integrate.cpp

```diff
@@ -272,51 +272,46 @@ namespace vcpkg::Commands::Integrate
                     default: Checks::unreachable(VCPKG_LINE_INFO);
                 }
             }
         }
-        bool should_install_system = true;
+
         std::error_code ec;
         std::string system_wide_file_contents = fs.read_contents(SYSTEM_WIDE_TARGETS_FILE, ec);
         if (!ec)
         {
             auto opt = find_targets_file_version(system_wide_file_contents);
             if (opt.value_or(0) >= 1)
             {
-                should_install_system = false;
+                return true;
             }
         }
 
-        if (should_install_system)
-        {
-            const auto tmp_dir = fs.create_or_get_temp_directory(VCPKG_LINE_INFO);
-            const auto sys_src_path = tmp_dir / "vcpkg.system.targets";
-            fs.write_contents(sys_src_path, create_system_targets_shortcut(), VCPKG_LINE_INFO);
-
-            const std::string param = fmt::format(R"(/c "mkdir "{}" & copy "{}" "{}" /Y > nul")",
-                                                  SYSTEM_WIDE_TARGETS_FILE.parent_path(),
-                                                  sys_src_path,
-                                                  SYSTEM_WIDE_TARGETS_FILE);
-            const ElevationPromptChoice user_choice = elevated_cmd_execute(param);
-            fs.remove_all(tmp_dir, VCPKG_LINE_INFO);
-            switch (user_choice)
-            {
-                case ElevationPromptChoice::YES: break;
-                case ElevationPromptChoice::NO:
-                    msg::println_warning(msgIntegrationFailed);
-                    Checks::exit_fail(VCPKG_LINE_INFO);
-                default: Checks::unreachable(VCPKG_LINE_INFO);
-            }
+        const auto tmp_dir = fs.create_or_get_temp_directory(VCPKG_LINE_INFO);
+        const auto sys_src_path = tmp_dir / "vcpkg.system.targets";
+        fs.write_contents(sys_src_path, create_system_targets_shortcut(), VCPKG_LINE_INFO);
+
+        const std::string param = fmt::format(R"(/c "mkdir "{}" & copy "{}" "{}" /Y > nul")",
+                                              SYSTEM_WIDE_TARGETS_FILE.parent_path(),
+                                              sys_src_path,
+                                              SYSTEM_WIDE_TARGETS_FILE);
+        elevated_cmd_execute(param);
+        fs.remove_all(tmp_dir, VCPKG_LINE_INFO);
+
+        Checks::msg_check_exit(VCPKG_LINE_INFO,
+                               !fs.exists(SYSTEM_WIDE_TARGETS_FILE, IgnoreErrors{}),
+                               msgSystemTargetsInstallFailed,
+                               msg::path = SYSTEM_WIDE_TARGETS_FILE);
 
-            Checks::msg_check_exit(VCPKG_LINE_INFO,
-                                   fs.exists(SYSTEM_WIDE_TARGETS_FILE, IgnoreErrors{}),
-                                   msgSystemTargetsInstallFailed,
-                                   msg::path = SYSTEM_WIDE_TARGETS_FILE);
+        if (!fs.exists(SYSTEM_WIDE_TARGETS_FILE, IgnoreErrors{}))
```

Comment:
This `if()` will never fail -- because `msg_check_exit()` immediately before will terminate the program.

You probably wanted to remove the `msg_check_exit()` call.

Reply:
You're right, that shouldn't be there. Sorry about that.